### PR TITLE
6916: Populate page from ordapi

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/pages/sections/supplier/supplier/controller.js [hunk 1]
```diff
@@ -27,11 +27,31 @@ const formatPutData = data => ({
...
-  const getSupplierDataEndpoint = getEndpoint({ endpointLocator: 'getSupplier', options: { supplierId } });
-  const supplierData = await getData({ endpoint: getSupplierDataEndpoint, accessToken, logger });
+  const ordapiSupplierDataEndpoint = getEndpoint({ endpointLocator: 'getOrdapiSupplier', options: { orderId } });
+  const ordapiSupplierData = await getData({
+    endpoint: ordapiSupplierDataEndpoint, accessToken, logger,
+  });
 
-  const context = getContext({ orderId, supplierData });
-  return context;
+  if (ordapiSupplierData.name) {
```
Comment (alibombaye) [inline app/pages/sections/supplier/supplier/controller.js:35]: can we have a double guard here `ordapiSupplierData && ordapiSupplierData.name`

File: app/pages/sections/supplier/supplier/controller.test.js [hunk 2]
```diff
@@ -10,33 +10,103 @@ jest.mock('./contextCreator', () => ({
...
+      });
+
+      it('should throw error', async () => {
+        getData.mockResolvedValueOnce({});
+        try {
+          await getSupplierPageContext({ orderId, accessToken });
+        } catch (err) {
+          expect(err).toEqual(new Error());
+        }
+      });
```
Comment (alibombaye) [inline app/pages/sections/supplier/supplier/controller.test.js:109]: 👍

File: app/pages/sections/supplier/supplier/ui.test.js [hunk 3]
```diff
@@ -41,16 +41,37 @@ const supplierData = {
   },
 };
 
-const mocks = () => {
-  nock(solutionsApiUrl)
-    .get('/api/v1/suppliers/supplier-1')
-    .reply(200, supplierData);
+const supplierDataFromDapi = {
```
Comment (alibombaye) [inline app/pages/sections/supplier/supplier/ui.test.js:44]: `supplierDataFromDapi` should this be Bapi?

File: app/pages/sections/supplier/supplier/ui.test.js [hunk 4]
```diff
@@ -143,13 +164,31 @@ test('should render supplier name', async (t) => {
...
 
+  const heading = Selector('h3[data-test-id="supplier-name-heading"]');
+  const text = Selector('div[data-test-id="supplier-name"]');
+
+  await t
+    .expect(heading.exists).ok()
+    .expect(await extractInnerText(heading)).eql(content.supplierNameHeading)
+    .expect(text.exists).ok()
+    .expect(await extractInnerText(text)).eql(supplierDataFromDapi.name);
+});
```
Comment (alibombaye) [inline app/pages/sections/supplier/supplier/ui.test.js:186]: This is nice. This is something we have missing in call-off to check we show the details from one api or the other.. 
Just the api name shouldn't be Dapi here

I'll add a tech debt task to do the same for call-off

File: app/pages/sections/supplier/supplier/controller.js (hunk 1 above)
Comment (kat1906) [reply app/pages/sections/supplier/supplier/controller.js:35]: yeah I saw this on yours

File: app/pages/sections/supplier/supplier/ui.test.js (hunk 3 above)
Comment (kat1906) [reply app/pages/sections/supplier/supplier/ui.test.js:44]: indeed it should - too many -apis!

**Pull request merged**